src/Servers/Kestrel/Core/src/Internal/Infrastructure/HttpCharacters.cs

```diff
@@ -198,5 +199,23 @@ public static int IndexOfInvalidFieldValueChar(string s)
 
             return -1;
         }
+
+        // Disallows control characters but allows extended characters > 0x7F
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static int IndexOfInvalidFieldValueCharExtended(string s)
+        {
+            var fieldValue = _fieldValue;
+
+            for (var i = 0; i < s.Length; i++)
+            {
+                var c = s[i];
+                if (c < (uint)fieldValue.Length && !fieldValue[c])
```

Comment:
Is this condition correct?

We want to disallow control characters (given by `fieldValue`), and allow characters > 0x7F.
Say we get 'A' (ASCII: 65), then it looks like `if (65 < 128 && ...)` which is true and the `i` is returned. Expected is `-1` as return value.

I see CI looks fine, so it's either me missing something or there's a test missing.

Reply:
I think it's right.
`if (c < (uint)fieldValue.Length && !fieldValue[c])` where c is `A` is `if (65 < 128 && !fieldValue[65])` where `fieldValue[65]` is `true` so `if (65 < 128 && !true)` so the if fails and the loop does not exit early.

Reply:
You're right.

> so it's either me missing something

It's me...`&&` and not `||` which my brain thought about. 
Thanks for clarification!